pkg/cmd/create_test.go

```diff
@@ -60,27 +64,148 @@ func TestCreateCmd(t *testing.T) {
 }
 
 func TestCreateStarterCmd(t *testing.T) {
+	tests := []struct {
+		name            string
+		chartAPIVersion string
+		useAbsolutePath bool
+		expectedVersion string
+	}{
+		{
+			name:            "v2 with relative starter path",
+			chartAPIVersion: "",
+			useAbsolutePath: false,
+			expectedVersion: chart.APIVersionV2,
+		},
+		{
+			name:            "v2 with absolute starter path",
+			chartAPIVersion: "",
+			useAbsolutePath: true,
+			expectedVersion: chart.APIVersionV2,
+		},
+		{
+			name:            "v3 with relative starter path",
+			chartAPIVersion: "v3",
+			useAbsolutePath: false,
+			expectedVersion: chartv3.APIVersionV3,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Chdir(t.TempDir())
+			ensure.HelmHome(t)
+			defer resetEnv()()
+
+			// Enable feature gate for v3 charts
+			if tt.chartAPIVersion == "v3" {
+				t.Setenv(string(gates.ChartV3), "1")
+			}
+
+			cname := "testchart"
+
+			// Create a starter using the appropriate chartutil
+			starterchart := helmpath.DataPath("starters")
+			os.MkdirAll(starterchart, 0o755)
+			var err error
+			var dest string
+			if tt.chartAPIVersion == "v3" {
+				dest, err = chartutilv3.Create("starterchart", starterchart)
+			} else {
+				dest, err = chartutil.Create("starterchart", starterchart)
+			}
+			if err != nil {
+				t.Fatalf("Could not create chart: %s", err)
+			}
+			t.Logf("Created %s", dest)
+
+			tplpath := filepath.Join(starterchart, "starterchart", "templates", "foo.tpl")
+			if err := os.WriteFile(tplpath, []byte("test"), 0o644); err != nil {
+				t.Fatalf("Could not write template: %s", err)
+			}
+
+			// Build the command
+			starterArg := "starterchart"
+			if tt.useAbsolutePath {
+				starterArg = filepath.Join(starterchart, "starterchart")
+			}
+			cmd := fmt.Sprintf("create --starter=%s", starterArg)
+			if tt.chartAPIVersion != "" {
```

Comment:
Should this follow the convention used elsewhere with `if tt.chartAPIVersion == "v3"` for the same thing?